TypeError: Cannot read property 'failedTypeParameterIndex' of undefined

I'm getting this exception with TypeScript 1.5.0-beta when I'm creating a class which doesn't satisfy the constraits of the generic type.

```
C:\Projects\my_project\node_modules\typescript\bin\tsc.js:14007
                    ts.Debug.assert(resultOfFailedInference.failedTypeParamete
                                                           ^
TypeError: Cannot read property 'failedTypeParameterIndex' of undefined
    at resolveCall (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:14007:60)
    at resolveCallExpression (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:14088:28)
    at getResolvedSignature (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:14170:47)
    at checkCallExpression (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:14186:29)
    at checkExpressionWorker (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:14875:28)
    at checkExpressionOrQualifiedName (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:14826:42)
    at checkExpression (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:14818:20)
    at checkExpressionStatement (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:15785:13)
    at checkSourceElement (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:16946:28)
    at Object.forEach (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:37:30)
    at checkBlock (C:\Projects\my_project\node_modules\typescript\bin\tsc.js:15551:16)
```

Unfortunately, I'm not able to create a small reproducable test case. It should be similar to this:
```
interface A {
    foo: any;
    bar: any;
}

interface C {
    prop: any;
}

interface B extends C {
    foo: any;
}

module Test {
    export class Foo<T extends A> {
    }
}

class Bar extends Test.Foo<B> {
}
```

but that reports the error correctly.

If I comment out lines 14007...14011 in my application's build, the error is reported correctly:
```
C:/Projects/my_project/lib/data_storage.ts(282,45): error TS2344: Type 'FooConfig' does not satisfy the constraint 'BarConfig'.
  Types of property 'foo' are incompatible.
    Type '{ service: { id: string; fooUri: string; barUri: string; }; otherProperty: { barUri: s...' is not assignable to type '{ service: { name: string; type: string; id: string; fooUri: string; barUri: string; ...'.
      Types of property 'service' are incompatible.
        Type '{ id: string; fooUri: string; barUri: string; }' is not assignable to type '{ name: string; type: string; id: string; fooUri: string; barUri: string; }'.
          Property 'name' is missing in type '{ id: string; fooUri: string; barUri: string; }'.
```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -7275,8 +7275,8 @@ module ts {
                 checkApplicableSignature(node, args, candidateForArgumentError, assignableRelation, /*excludeArgument*/ undefined, /*reportErrors*/ true);
             }
             else if (candidateForTypeArgumentError) {
-                if (!isTaggedTemplate && (<CallExpression>node).typeArguments) {
-                    checkTypeArguments(candidateForTypeArgumentError, (<CallExpression>node).typeArguments, [], /*reportErrors*/ true)
+                if (!isTaggedTemplate && typeArguments) {
+                    checkTypeArguments(candidateForTypeArgumentError, typeArguments, [], /*reportErrors*/ true)
                 }
                 else {
                     Debug.assert(resultOfFailedInference.failedTypeParameterIndex >= 0);

--- a/tests/baselines/reference/genericTypeConstraints.errors.txt
+++ b/tests/baselines/reference/genericTypeConstraints.errors.txt
@@ -1,0 +1,24 @@
+tests/cases/compiler/genericTypeConstraints.ts(9,27): error TS2344: Type 'FooExtended' does not satisfy the constraint 'Foo'.
+  Property 'fooMethod' is missing in type 'FooExtended'.
+tests/cases/compiler/genericTypeConstraints.ts(9,31): error TS2344: Type 'FooExtended' does not satisfy the constraint 'Foo'.
+
+
+==== tests/cases/compiler/genericTypeConstraints.ts (2 errors) ====
+    class Foo {
+        fooMethod() {}
+    }
+    
+    class FooExtended { }
+    
+    class Bar<T extends Foo> { }
+    
+    class BarExtended extends Bar<FooExtended> {
+                              ~~~~~~~~~~~~~~~~
+!!! error TS2344: Type 'FooExtended' does not satisfy the constraint 'Foo'.
+!!! error TS2344:   Property 'fooMethod' is missing in type 'FooExtended'.
+                                  ~~~~~~~~~~~
+!!! error TS2344: Type 'FooExtended' does not satisfy the constraint 'Foo'.
+        constructor() {
+            super();
+        }
+    }

--- a/tests/baselines/reference/genericTypeConstraints.js
+++ b/tests/baselines/reference/genericTypeConstraints.js
@@ -1,0 +1,45 @@
+//// [genericTypeConstraints.ts]
+class Foo {
+    fooMethod() {}
+}
+
+class FooExtended { }
+
+class Bar<T extends Foo> { }
+
+class BarExtended extends Bar<FooExtended> {
+    constructor() {
+        super();
+    }
+}
+
+//// [genericTypeConstraints.js]
+var __extends = (this && this.__extends) || function (d, b) {
+    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
+    function __() { this.constructor = d; }
+    __.prototype = b.prototype;
+    d.prototype = new __();
+};
+var Foo = (function () {
+    function Foo() {
+    }
+    Foo.prototype.fooMethod = function () { };
+    return Foo;
+})();
+var FooExtended = (function () {
+    function FooExtended() {
+    }
+    return FooExtended;
+})();
+var Bar = (function () {
+    function Bar() {
+    }
+    return Bar;
+})();
+var BarExtended = (function (_super) {
+    __extends(BarExtended, _super);
+    function BarExtended() {
+        _super.call(this);
+    }
+    return BarExtended;
+})(Bar);

--- a/tests/cases/compiler/genericTypeConstraints.ts
+++ b/tests/cases/compiler/genericTypeConstraints.ts
@@ -1,0 +1,13 @@
+class Foo {
+    fooMethod() {}
+}
+
+class FooExtended { }
+
+class Bar<T extends Foo> { }
+
+class BarExtended extends Bar<FooExtended> {
+    constructor() {
+        super();
+    }
+}

